Header header
Float64MultiArray csi1_real
Float64MultiArray csi1_image
Float64MultiArray csi2_real
Float64MultiArray csi2_image
bool check_csi
